### PR TITLE
STAF-35: add slide in menu from the sides

### PR DESCRIPTION
**Pull request by RomanLavrunov**

### CODE REVIEW

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: src/core/components/SlideInProvider/SlideInProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export const enum SLIDE_FROM_SIDE {
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+export const SlideInProvider = () => {
+  const slideInBarsConfigsArray = useAppSelector(({ core }) => core.slideInBarConfigs);
+  const [Component, setComponent] = useState<JSX.Element | null>(null);
+  const [slideFrom, setSlideFrom] = useState(SLIDE_FROM_SIDE.LEFT);
+  const title = slideFrom === 'left' ? 'Menu' : 'Profile';
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/core/components/SlideInProvider/SlideInProvider.tsx]: fixed

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: src/config/slideInMenuConfig.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { MainMenuBlock } from '../core/components/Menus/MainMenuBlock';
+import { ProfileMenuBlock } from '../user/components/ProfileMenuBlock';
+import { ProtectedRoutesEnum } from './routes/routes';
+
+export type MenuBlockType = {
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/config/slideInMenuConfig.ts]: Name changed.

File: src/core/components/Menus/MenuItemList/MenuItemList.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  componentName: string;
+}
+
+export const MenuItemList = ({ itemList, componentName }: MenuItemListProps): JSX.Element => {
+  return (
+    <ul className="menu-item-list">
+      {itemList.map((item: MenuBlockType) => (
+        <Link to={item.url} key={item.id}>
+          <div role="button" className="menu-list-item ripple">
+            <span className="menu-list-item-title">{`${t(`core.slideInBar.${componentName}.${item.id}`)}`}</span>
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/core/components/Menus/MenuItemList/MenuItemList.tsx:18]: Comment added!

File: src/assets/theme/_variables.scss [hunk 4]
```diff
@@ -20,3 +16,18 @@
...
+$secondaryColorContrast: var(--secondary-color-contrast);
+
+$breakpoints: (
+  bp-320: 320px,
+  bp-360: 360px,
+  bp-480: 480px,
+  bp-768: 768px,
+  bp-1024: 1024px,
+  bp-1200: 1200px,
+  bp-1400: 1400,
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/assets/theme/_variables.scss]: Fixed!

**Review by devmc-ee — APPROVED** [review]

**Pull request merged**